### PR TITLE
Fix static scan issues

### PR DESCRIPTION
**Pull request by xianju6x**
Tracked-On: OAM-110031

### CODE REVIEW

File: crashlog/fsutils.h [hunk 1]
```diff
@@ -89,6 +89,7 @@ static inline int dir_exists(const char *dirpath) {
 
     dir = opendir(dirpath);
     if (dir != NULL)
+        closedir(dirpath);
```
Comment (jingdlu) [inline crashlog/fsutils.h:92]: You may need to use { } for closedir and return 1.

File: crashlog/crashutils.c [hunk 2]
```diff
@@ -164,13 +164,13 @@ unsigned long long get_uptime(int refresh, int *error)
...
 static int find_system_last_kmsg(char source[], int source_length) {
     struct dirent *entry;
-    DIR *dir = opendir(DROPBOX_DIR);
     int file_exist = 0;
 
     if (source == NULL) {
         LOGE("source is NULL.\n");
         return file_exist;
     }
+    DIR *dir = opendir(DROPBOX_DIR);
```
Comment (jingdlu) [inline crashlog/crashutils.c:173]: why need to move opendir here?

File: crashlog/crashutils.c [hunk 3]
```diff
@@ -808,13 +808,17 @@ int process_info_and_error(char *filename, char *name) {
         snprintf(path, sizeof(path),"%s/%s", filename,tmp_data_name);
         snprintf(destion,sizeof(destion),"%s/%s", dir, tmp_data_name);
         do_copy_tail(path, destion, 0);
-        remove(path);
```
Comment (jingdlu) [inline crashlog/crashutils.c:811]: Why need to check the return value of remove()? What's the consequence?

File: crashlog/usercrash.c [hunk 4]
```diff
@@ -116,7 +116,12 @@ static int priv_process_usercrash_event(struct watch_entry *entry, struct inotif
             do_log_copy(eventname, dir, get_current_time_short(1), APLOG_TYPE);
             break;
         case HPROF_TYPE:
-            remove(path);
+            if (remove(path) == -1) {
+                LOGE("Failed to remove path %s\n", path);
+                free(key);
+                free(dir);
+                return -1;
```
Comment (jingdlu) [inline crashlog/usercrash.c]: If you return here, other code logic may not be executed like raise_event.

File: crashlog/crashutils.c (hunk 2 above)
Comment (xianju6x) [reply crashlog/crashutils.c:173]: Variable dir going out of scope leaks the storage it points to at line 171.

File: crashlog/crashutils.c (hunk 3 above)
Comment (xianju6x) [reply crashlog/crashutils.c:811]: CID 206964: (#2 of 2): Unchecked return value from library (CHECKED_RETURN)
5. check_return Calling remove(path) without checking return value. This library function may fail and return an error code. [Note: The source code implementation of the function has been overridden by a builtin model.]

File: crashlog/usercrash.c (hunk 4 above)
Comment (xianju6x) [reply crashlog/usercrash.c]: Ok, just check return value and print some logs

**Review by jingdlu — APPROVED** [review]

**Review by kwang13 — APPROVED** [review]

**Pull request merged**